sbi/inference/snpe/snpe_base.py

```diff
@@ -90,7 +90,7 @@ def __init__(
         self._neural_posterior = Posterior(
             algorithm_family="snpe",
             neural_net=density_estimator,
-            prior=prior,
+            prior=self._prior,
```

Comment:
It was a bug. The `Posterior` is created in `__init__` where the (user) prior is still visible. Instead of passing the processed `self._prior` it was passing the unprocessed (user) `prior` here.